src/goto-analyzer/static_verifier.cpp

```diff
@@ -27,6 +27,39 @@ struct static_verifier_resultt
   irep_idt function_id;
 };
 
+void static_verifier(
+  const abstract_goto_modelt &abstract_goto_model,
+  const ai_baset &ai,
+  propertiest &properties)
+{
+  auto const ns = namespacet{abstract_goto_model.get_symbol_table()};
+  for(auto &property : properties)
+  {
+    auto &property_status = property.second.status;
+    auto const &property_location = property.second.pc;
+    auto condition = property_location->get_condition();
+    auto const predecessor = ai.abstract_state_before(property_location);
+    auto const &domain = *predecessor;
+    domain.ai_simplify(condition, ns);
+    if(condition.is_true())
+    {
+      property_status = property_statust::PASS;
+    }
+    else if(condition.is_false())
+    {
+      property_status = property_statust::FAIL;
+    }
+    else if(domain.is_bottom())
+    {
+      property_status = property_statust::NOT_REACHABLE;
+    }
+    else
+    {
+      property_status = property_statust::UNKNOWN;
```

Comment:
How can this happen? If the location is never assigned a value then it should be `NOT_REACHED`.

Reply:
This means the condition could not be simplified to true or false using domain information (so it's neither definite fail or success) and the domain is _not_ bottom so the location is reachable, therefore it is unknown.

Reply:
It would be helpful to have this commentary on the various states in the code as well, and not just in the PR :-)

Reply:
I added some comments explaining what was going on